src/txn/txn_recover.c

```diff
@@ -452,6 +452,69 @@ __recovery_set_oldest_timestamp(WT_RECOVERY *r)
     return (0);
 }
 
+/*
+ * __recovery_set_checkpoint_snapshot --
+ *     Set the checkpoint snapshot details as retrieved from the metadata file.
+ */
+static int
+__recovery_set_checkpoint_snapshot(WT_RECOVERY *r)
+{
+    WT_CONFIG list;
+    WT_CONFIG_ITEM cval;
+    WT_CONFIG_ITEM k;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    uint8_t counter;
+    char *sys_config;
+
+    sys_config = NULL;
+    session = r->session;
+    conn = S2C(session);
+    counter = 0;
+
+    /*
+     * Read the system checkpoint information from the metadata file and save the snapshot related
+     * details of the last checkpoint for later query. This gets saved in the connection.
+     */
+
+    /* Search the metadata for the system information. */
+    WT_ERR_NOTFOUND_OK(
+      __wt_metadata_search(session, WT_SYSTEM_CKPT_SNAPSHOT_URI, &sys_config), false);
+    if (sys_config != NULL) {
+        WT_CLEAR(cval);
+        WT_ASSERT(session,
+          __wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MIN, &cval) == 0);
+        if (cval.len != 0)
+            conn->recovery_ckpt_snap_min = (uint64_t)cval.val;
+
+        WT_ASSERT(session,
+          __wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MAX, &cval) == 0);
+        if (cval.len != 0)
+            conn->recovery_ckpt_snap_max = (uint64_t)cval.val;
+
+        WT_ASSERT(session,
+          __wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_COUNT, &cval) == 0);
+        if (cval.len != 0)
+            conn->recovery_ckpt_snapshot_count = (uint32_t)cval.val;
+
+        WT_ASSERT(
```

Comment:
Will this fail? Is it OK to have a snapshot with only a min/max?

Reply:
My preference is for checking the expected content at the end of all the parsing, rather than interspersed with the code.

Reply:
Fixed. Updated the code to assert at the end of parsing to check if the metadata has expected value.